packages/plugin-pnpm/sources/PnpmLinker.ts

```diff
@@ -79,9 +79,12 @@ export class PnpmLinker implements Linker {
 
 class PnpmInstaller implements Installer {
   private readonly asyncActions = new miscUtils.AsyncActions(10);
+  private readonly indexFolderPromise: Promise<PortablePath>;
 
   constructor(private opts: LinkOptions) {
-    // Nothing to do
+    this.indexFolderPromise = setupCopyIndex(xfs, {
+      indexPath: ppath.join(opts.project.configuration.get(`globalFolder`), `index` as Filename),
+    });
```

Comment:
I don't like that the constructor has side-effects, can't this happen later when it's actually needed?

Reply:
I did it early to avoid having to either do IO calls for each package, or having to keep a boolean state (which I feel like would add complexity to avoid a situation - a pnpm install on an empty project without any other install - that doesn't happen much, and doesn't cost much even if it happens).

Reply:
I would rather have that boolean state, this will do 257 `mkdir` calls **every time** `PnpmInstaller` is constructed, regardless of which linker is in use and if the directories already exists.